src/modules/Channel/components/MessageFeedbackModal/hooks/useKeyDownV2.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useLayoutEffect } from 'react';
+
+type KeyDownCallbackMap = Record<string, (event: React.KeyboardEvent<HTMLDivElement>) => void>;
+
+interface UseKeyDownV2Props {
+  ref: React.RefObject<HTMLDivElement>;
+  keyDownCallbackMap: KeyDownCallbackMap;
+}
+
+export function useKeyDownV2(props: UseKeyDownV2Props): React.KeyboardEventHandler<HTMLDivElement> {
```

Comment:
How about simplifying the parameter interface of the hook for easier usage?

```ts
function useKeydown(ref: React.RefObject<HTMLDivElement>, listener: KeyDownCallbackMap) { ... }
```

```ts
// AS-IS
useKeydown({
  ref: elemRef,
  keyDownCallbackMap: {
    Escape: () => { ... },
    ArrowLeft: () => { ... },
    ArrowRight: () => { ... },
    Enter: () => { ... },
  }
})

// TO-BE
useKeydown(elemRef, {
  Escape: () => { ... },
  ArrowLeft: () => { ... },
  ArrowRight: () => { ... },
  Enter: () => { ... },
})
```

Reply:
SG2M!

Reply:
ah actually we are trying to benefit by using
```
const callback = keyDownCallbackMap[event.key];
callback?.(event);
```
so lets keep as is

Reply:
Oh, the object form will be retained.

```ts
// AS-IS
const callback = params.keyDownCallbackMap[event.key];
callback?.(event);

// TO-BE
const callback = keyDownCallbackMap[event.key]; // callbacks[event.key]
callback?.(event);
```

Reply:
did you mean to prefer
```
const callback = props[event.key];
```
?

Reply:
Also we want to use `keyDownCallbackMap` as dependency for `useCallback` so I think we had better keep as is
```
interface UseKeyDownProps {
  ref: React.RefObject<HTMLDivElement>;
  keyDownCallbackMap: KeyDownCallbackMap;
}
```

Reply:
Ah, the reason I suggested it is because I felt it unnecessary to specify the `keyDownCallbackMap` key when using the hook.

```tsx
// AS-IS
useKeydown({
  ref: elemRef,
  keyDownCallbackMap: { // -> here
    Escape: () => { ... },
    ArrowLeft: () => { ... },
    ArrowRight: () => { ... },
    Enter: () => { ... },
  }
})

// TO-BE
useKeydown(elemRef, {
  Escape: () => { ... },
  ArrowLeft: () => { ... },
  ArrowRight: () => { ... },
  Enter: () => { ... },
})
```

Regarding this, there are no suggestions for changes in the internal logic.

Reply:
gotcha!